ktor-client/ktor-client-core/common/src/io/ktor/client/request/HttpRequest.kt

```diff
@@ -85,7 +85,7 @@ class HttpRequestBuilder : HttpMessageBuilder {
      * A deferred used to control the execution of this request.
      */
     @KtorExperimentalAPI
-    public var executionContext: Job = Job()
+    public var executionContext: Job = SupervisorJob()
```

Comment:
I would like to see more tests with `request.await()`: the case where you start request and awaiting completion on context. Especially cases with redirect, timeout, and cancel.

Reply:
A lot of different coroutines are started under the job. Are we sure that it's safe to avoid auto crash handling?

Reply:
That thing we can test actually. For instance, add a test that cancels body or launch and cancel child coroutine

Reply:
@cy6erGn0m afaik, `SupervisorJob` doesn't avoid auto crash handling but keep parent job active for reuse. All exceptions are still propagating to the caller. Do I miss anything?

Added test for failure in child coroutine to verify that caller will receive an exception and the parent job is still active.